### PR TITLE
[Merged by Bors] - Update and restructure documentation

### PR DESCRIPTION
**Pull request by lfrancke**
## Description

This updates various bits of the documentation a it.
In particular it explains in more detail which databases we support and which issues can occur with newer PostgreSQL versions.

## Review Checklist
- [ ] Documentation added (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by lfrancke** [comment]
@semmjon Does this explanation help?

**Review by maltesander — APPROVED** [review]
LGTM.

File: docs/modules/ROOT/pages/installation.adoc [hunk 1]
```diff
@@ -57,10 +57,10 @@ docker run \
...
 
-This operator is written in Rust and is developed against the latest stable Rust release (1.57 at the time of writing).
+This operator is written in Rust and is developed against the latest stable Rust release (1.60 at the time of writing).
 
-[source]
+[source,bash]
 ----
-stackable-hive-operator crd | kubectl apply -f -
-stackable-hive-operator run
+cargo run stackable-hive-operator crd | kubectl apply -f -
```
Comment (fhennig) [inline docs/modules/ROOT/pages/installation.adoc:64]: On my machine, this doesn't actually run. I think what you mean is:

    cargo run crd | kubectl apply -f -
    cargo run run

**Review by fhennig — CHANGES_REQUESTED** [review]
Just one thing, other than that it looks good to me :ok_hand:

File: docs/modules/ROOT/pages/usage.adoc [hunk 2]
```diff
@@ -61,20 +143,27 @@ To create a single node Apache Hive Metastore (v2.3.9) cluster with derby and S3
...
-WARNING: PostgreSQL versions 11 or higher are currently not supported (see https://github.com/stackabletech/hive-operator/issues/167):
+To create a single node Apache Hive Metastore using PostgreSQL, deploy a PostgreSQL instance via helm.
+
+[sidebar]
+PostgreSQL introduced a new way to encrypt its passwords in version 10.
+This is called `scram-sha-256` and has been the default as of PostgreSQL 14.
+Unfortunately, Hive up until the latest 3.3.x version ships with JDBC drivers that do _not_ support this method.
+You might see an error message like this:
+`The authentication type 10 is not supported.`
+If this is the case please either use an older PostgreSQL version or change its https://www.postgresql.org/docs/current/runtime-config-connection.html#GUC-PASSWORD-ENCRYPTION[`password_encryption`] setting to `md5`.
```
Comment (lfrancke) [inline docs/modules/ROOT/pages/usage.adoc:155]: Should mention the docs for the jdbc driver

File: docs/modules/ROOT/pages/usage.adoc [hunk 3]
```diff
@@ -61,20 +143,27 @@ To create a single node Apache Hive Metastore (v2.3.9) cluster with derby and S3
...
 
-[source]
+This installes PostgreSQL in version 10 to work around the issue mentioned above:
```
Comment (lfrancke) [inline docs/modules/ROOT/pages/usage.adoc:157]: Typo "installs"

File: docs/modules/ROOT/pages/installation.adoc (hunk 1 above)
Comment (lfrancke) [reply docs/modules/ROOT/pages/installation.adoc]: Thank you! Fixed now.

**Review by fhennig — APPROVED** [review]
LGTM

**Comment by lfrancke** [comment]
bors merge

**Pull request closed without merging**